content/cloud/linking-applications-to-resources-using-labels.md

```diff
@@ -0,0 +1,152 @@
+
+---
+
+- [Features of Links and Labels](#features-of-links-and-labels)
+- [Working With Links and Labels](#working-with-links-and-labels)
+- [Next Steps](#next-steps)
+
+Many Spin applications require data storage between invocations. Fermyon Cloud offers two methods for this: key-value storage and NoOps SQL databases. However, the relationship between applications and databases isn't strictly one-to-one. A single database might be used by multiple applications, while an application could utilize several databases. There might even be a need to maintain a database that isn't utilized by any applications, perhaps for analytics or compliance reasons. All of these scenarios can occur as applications are independently deployed, upgraded, and retired, without any particular application needing to be aware of how others might be using the same databases.
+
+To facilitate the seamless transition of applications between different environments, such as from a local development environment to Fermyon Cloud, Spin applications don't reference databases by their physical location, like a connection string. Instead, they use a label. A label is an abstract identifier, such as "default," "user-accounts," or "transactions." The environment determines how this label is mapped to a physical database.
+
+In the Spin development environment, that mapping is handled by the [runtime configuration file](/spin/dynamic-configuration#runtime-configuration).  In Fermyon Cloud, it's managed through _links_.
+
+> Link management is currently available only for NoOps SQL databases. Fermyon Cloud currently allows applications only the "default" key-value store, which is private to the application and shares the application's lifespan. Key-value link management is planned for a future release of Fermyon Cloud.
+
+This may sound abstract, so let's walk through a specific example:
+
+This is where Fermyon Cloud **links** and **labels** are helpful. As a Spin application developer, you provide a label whenever you fill out the name for a NoOps SQL database in the component manifest. Let's walk through a specific example together:
+
+```toml
+// --snip--
+name = "todo-app"
+[component]
+id = "todo"
+sqlite_databases = ["data"]
+// --snip--
+```
+
+The string "data" is the **label** that the Spin application uses to reference the NoOps SQL Database. At application deployment time, we have to make a decision about how to map that label to an actual database.  The `spin cloud deploy` command gives us a couple of choices:
+- Use an existing database and link app to it
+- Create a new database and link the app to it
+
+In this example below, we are going to **link** the app to a specific NoOps SQL Database instance using the label. To do this, we choose the first option (that `spin deploy` gives us). We make our selection by using the up/down arrow keys and then `return` to continue:
```

Comment:
```suggestion
In this example below, we are going to **link** the app to an existing NoOps SQL Database instance using the label. To do this, we choose the first option (that `spin deploy` gives us). We make our selection by using the up/down arrow keys and then `return` to continue:
```

It always links to a "specific" database - "in this example" implies we are talking about the decision being made in the example, which us "link to existing" vs "create and link to new."